forte/data/data_store.py

```diff
@@ -10,69 +10,77 @@
+        Different `entry lists` are ordered by the `type_id` of the type of entries
+        stored in this list.
 
         Entries are stored as `entry data` in each entry list.
         Each `entry data` in the `entry list` is represented by a list of attributes,
-        For example, annotation type entry has the following format:
+        For example, an annotation type entry has the following format:
         [<begin>, <end>, <tid>, <entry_type>, <attr_1>, <attr_2>, ..., <attr_n>].
-
-        The first four fields are compulsory for every `entry data`. They are always in
-        the order of `begin`, `end`, `tid` and `entry type`, .
-        `Begin` and `end` are the begin and end character index of this entry in the text.
-        `Entry type` is the type of this entry. `tid` is a unique id of every entry,
-        usually generated by uuid.uuid4().
-        Each entry type has a fixed field of attributes.
-        E.g. an annotation-type `entry data` with type Document has the following structure:
-        [<begin>, <end>, <tid>, <entry_type>, <document_class>, <sentiment>, <classifications>]
-
-        Different entries are sorted by the `begin` attribute of the entry.
-        If two entries have the same begin position, then we used the `end` attribute to sort them.
+        A group type entry has the following format:
+        [<member_type>, <[members_tid_list]>, <tid>, <entry_type>, <attr_1>,
+            <attr_2>, ..., <attr_n>].
+        A link type entry has the following format:
+        [<parent_tid>, <child_tid>, <tid>, <entry_type>, <attr_1>, <attr_2>, ..., <attr_n>].
+
+        The first four fields are compulsory for every `entry data`. The third
+        and fourth fields are always `tid` and `entry_type`, but the first and
+        second fields can change across different types of entries.
+        For example, first four fields of annotations entries are always in
+        the order of `begin`, `end`, `tid` and `entry_type`. `Begin` and `end`,
+        which are compulsory for annotations entries, represent the begin and
+        end character indices of entries in the text.
+        `entry_type` is the fully qualified name of every entry. `tid` is an unique
+        id of every entry, usually generated by uuid.uuid4().
+        Each `entry_type` has a pre-defined ordered list of attributes, which is
+        determined by the ontology specification.
+        E.g. an annotation-type `entry data` with type `ft.onto.base_ontology.Document`
+        has the following structure:
+        [<begin>, <end>, <tid>, <entry_type>, <document_class>, <sentiment>, <classifications>].
+
+        Annotation entries are sorted by the first two attributes (`begin`, `end`)
+        of the entry.
+        The order of other entry types that's not Annotation-like, is arbitrary.
+
+        `__onto_file_path` is an optional argument, which could pass in an user
+        defined ontology file. The user defined ontology class should be able
```

Comment:
Instead of `The user defined ontology class`, should be specific that entry types defined in the provided file.

Reply:
yes will change